packages/java/parser-jvm-core/src/main/java/dev/hilla/parser/models/PropertyInfoReflectionModel.java

```diff
@@ -0,0 +1,55 @@
+package dev.hilla.parser.models;
+
+import java.util.Optional;
+
+import com.fasterxml.jackson.databind.introspect.BeanPropertyDefinition;
+
+final class PropertyInfoReflectionModel extends PropertyInfoModel {
+    private final BeanPropertyDefinition origin;
+
+    PropertyInfoReflectionModel(BeanPropertyDefinition origin,
+            ClassInfoModel owner) {
+        super(owner);
+        this.origin = origin;
+    }
+
+    @Override
+    public BeanPropertyDefinition get() {
+        return origin;
+    }
+
+    @Override
+    public boolean hasGetter() {
+        return origin.hasGetter();
+    }
+
+    @Override
+    public boolean isTransient() {
+        return !origin.hasGetter() && origin.getField().isTransient();
+    }
+
+    @Override
+    protected FieldInfoModel prepareField() {
+        return FieldInfoModel.of(origin.getField().getAnnotated());
+    }
+
+    @Override
+    protected Optional<MethodInfoModel> prepareGetter() {
```

Comment:
Do we really need to know about getters? Jackson properties should already deal with them.

Reply:
Well, they're just helper methods in case anyone wants to use field or getter directly. It has the same goal as `get()` method in any other model but works with models.